specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2024-04-01-preview/fluxconfiguration.json

```diff
@@ -0,0 +1,1822 @@
+      "type": "object",
+      "x-nullable": true,
+      "properties": {
+        "clientId": {
+          "description": "The client Id for authenticating a Managed Identity.",
+          "type": "string",
+          "x-nullable": true
+        }
+      }
+    },
+    "ManagedIdentityPatchDefinition": {
+      "description": "Parameters to authenticate using a Managed Identity.",
+      "type": "object",
+      "x-nullable": true,
+      "properties": {
+        "clientId": {
+          "description": "The client Id for authenticating a Managed Identity.",
+          "type": "string",
+          "x-nullable": true
+        }
+      }
+    },
+    "OCIRepositoryDefinition": {
+      "description": "Parameters to reconcile to the OCIRepository source kind type.",
+      "type": "object",
+      "x-nullable": true,
+      "properties": {
+        "url": {
+          "description": "The URL to sync for the flux configuration OCI repository.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "timeoutInSeconds": {
+          "description": "The maximum time to attempt to reconcile the cluster OCI repository source with the remote.",
+          "type": "integer",
+          "format": "int64",
+          "default": 600,
+          "x-nullable": true
+        },
+        "syncIntervalInSeconds": {
+          "description": "The interval at which to re-reconcile the cluster OCI repository source with the remote.",
+          "type": "integer",
+          "format": "int64",
+          "default": 600,
+          "x-nullable": true
+        },
+        "repositoryRef": {
+          "description": "The source reference for the OCIRepository object.",
+          "$ref": "#/definitions/OCIRepositoryRefDefinition"
+        },
+        "layerSelector": {
+          "description": "The layer to be pulled from the OCI artifact.",
+          "$ref": "#/definitions/LayerSelectorDefinition"
+        },
+        "verify": {
+          "description": "Verification of the authenticity of an OCI Artifact.",
+          "$ref": "#/definitions/VerifyDefinition"
+        },
+        "insecure": {
+          "description": "Specify whether to allow connecting to a non-TLS HTTP container registry.",
+          "type": "boolean",
+          "default": false
+        },
+        "useWorkloadIdentity": {
+          "description": "Specifies whether to use Workload Identity to authenticate with the OCI repository.",
+          "type": "boolean",
+          "default": false
+        },
+        "serviceAccountName": {
+          "description": "The service account name to authenticate with the OCI repository.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "tlsConfig": {
+          "description": "Parameters to authenticate using TLS config for OCI repository.",
+          "type": "object",
+          "$ref": "#/definitions/TlsConfigDefinition"
+        },
+        "localAuthRef": {
```

Comment:
>localAuthRef [](http://example.com/codeflow?start=9&length=12)

Consider a better name for this property, maybe clusterSecretName, localAuthentication, etc.

Reply:
We already use this property in some other sources in stable version. Changing it here would be inconsistent.